### ISSUE TITLE
`non-compilable` directory path name is confusing

### ISSUE THREAD

**Issue by nrmancuso** [checkstyle/checkstyle#13246]
The `non-compilable` directory name used in https://github.com/checkstyle/checkstyle/tree/master/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle is not intuitive. Nearly all the test resources there (inputs) ARE indeed compilable, just not at the language level of the rest of the project.

I think the `non-compilable` name IS appropriate for files that are actually not compilable for one reason or another (test case requires it, or some compiler is broken i.e. eclipse). However, these cases are few and far between. I think it would be much more intuitive to have a directory structure like this:
```

└── src
    └── test
        ├── resources
        ├── resources-jdk14
        ├── resources-jdk17
        ├── resources-jdk19
        └── resources-non-compilable

```

This structure would also make our scripting and CI checks easier, as well as migration of resources when we bump the project language level (simply merge directories instead of finding all files). We would also no longer need compilation comments in files.

**Comment by nrmancuso** [comment]
@stoyanK7 Please share any thoughts you have on this.

**Comment by stoyanK7** [comment]
Yes, `non-compilable` by itself was confusing because the tests are compilable as you said. I like the idea of having `resources-jdk14`, etc.. Makes it clear that it is compilable with a different version than what checkstyle is using.

Doesn't that mean that resources will have to be moved from `resources-jdk14` to `resources` if checkstyle decides to use it? Just wondering.

**Comment by nrmancuso** [comment]
Next migration will be to 17, at which point everything at or below that language level will just move to `resources`.

**Comment by romani** [comment]
There are some Inputs that are compilable by jdk but not compileable by Eclipse compiler :). So we moved it to such folder also.

I agree that now this folder is massive.
It use to let massive before java started more frequent releases, and we agreed to be more conservative on min jdk for us.
I am fine to change structure.

@rnveach , please approve issue if you agree.

**Comment by rnveach** [comment]
@nrmancuso 
Is there a reason we just don't include everything in the resource directory? Why do we need separation at all? It seems we are just making it 5x harder to find the specific class, or the location to add a new input.  You say `non-compilable` is confusing, but basically still leave it in place. :)

Unfortunately, we can't put all Java in one folder, otherwise we have to remove the local enforcement that the Java be compilable.
https://github.com/checkstyle/checkstyle/blob/fbbf38a0d9e3517cbea6e41ab95293d6ae20bbb6/pom.xml#L1357
I am actually not against removing this. CI can confirm input is really compilable.

Why don't we just have `resources-java`, `resources-java-future`, and `resources`. Anything that isn't Java or isn't compilable will just go into `resources`. This way we do away with `non-compilable` completely. We will probably need something that checks compilable java files aren't placed in `resources` by mistake.
